pkg/kubelet/types/pod_update.go

```diff
@@ -147,6 +147,21 @@ func IsCriticalPod(pod *v1.Pod) bool {
 	return IsCritical(pod.Namespace, pod.Annotations) || (pod.Spec.Priority != nil && IsCriticalPodBasedOnPriority(*pod.Spec.Priority))
 }
 
+// Preemptable returns true if preemptor pod can preempt preemptee pod:
+//   - If preemptor's is greater than preemptee's priority, it's preemptable (return true)
+//   - If preemptor (or its priority) is nil and preemptee bears the critical pod annotation key,
+//     preemptee can not be preempted (return false)
+//   - If preemptor (or its priority) is nil and preemptee's priority is greater than or equal to
+//     SystemCriticalPriority, preemptee can not be preempted (return false)
+func Preemptable(preemptor, preemptee *v1.Pod) bool {
+	if (preemptor != nil && preemptor.Spec.Priority != nil) &&
+		(preemptee != nil && preemptee.Spec.Priority != nil) {
+		return *(preemptor.Spec.Priority) > *(preemptee.Spec.Priority)
+	}
+
+	return !IsCriticalPod(preemptee)
```

Comment:
Do we support setting `CriticalPod` annotation with an explicit priority set?

Reply:
In admission controller (`Priority`), the `CriticalPod` is set to `SystemClusterCritical`.

Think about it just now, it's better to check `PodPriority` feature gate: if someone enable PodPriority before, and then disable it, we should check CriticalPod annotation instead of priority.